src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1538,6 +1541,41 @@ public void updateLatestRealtimeTask(
         }, null, false, listener);
     }
 
+    /**
+     * Update latest realtime task's init progress, estimated minutes left for initialization, state and error field.
+     *
+     * @param detectorId detector id
+     * @param totalUpdates total updates
+     * @param intervalMinutes interval minutes
+     * @param failure failure
+     */
+    public void updateLatestRealtimeTask(String detectorId, long totalUpdates, long intervalMinutes, Exception failure) {
```

Comment:
Do we save no matter what?  If yes, can we save as needed, similar to what we do in DetectionStateHandler.saveError?

Reply:
Not sure why, seems some code change not synced from my Mac to desktop due to VPN issue. Now you should be able to see the latest change.